### PR TITLE
feat(SDK): Provide a client.getPage method to fetch page, content, nav using graphQL

### LINKED ISSUE #30968: SDK: Provide a `client.getPage` method to fetch page, content, nav using graphQL
> [!CAUTION]
> THESE ARE BREAKING CHANGES AND SHOULD BE DOCUMENTED

### Parent Issue

#30943

### Overview of the Issue:

To enable a unified and flexible GraphQL querying system, we need to implement a method `client.getPage` that:

1. Merges a **base query** with user-defined **custom queries** using GraphQL fragments.
2. Ensures the base query includes all required fields for the Universal Visual Editor (UVE) to function correctly.
3. Provides flexibility by supporting optional properties for `page`, `content`, and `nav`, requiring at least one.
4. Processes the `_map` fields in the response to remove them from the output while maintaining their parsed values.

This method will be critical for both developers using the SDK and for internal UVE operations, ensuring consistent and optimized GraphQL usage.

### POC:

https://github.com/dotCMS/core/pull/30814

### Task

1. **Define the Base Page Query**
2. **Implement the** `client.getPage` **Method**:
    - Accept input for `page`, `content`, and `nav` queries.
    - Validate that at least one of `page`, `content`, or `nav`, is provided.
    - Merge the base query with user-provided custom queries using GraphQL fragments.
3. **Parse _map Fields:**
  - Remove `_map` fields from the GraphQL response.
  - Parse and include `_map` values in the output data structure.
4. **Document the Method**

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria
- [ ] The `client.getPage` method correctly merges the base query with custom queries using GraphQL fragments.
- [ ] The method validates input to ensure at least one of `page`, `content`, or `nav` is provided.
- [ ] `_map` fields are parsed and removed from the response.
- [ ] The method returns the expected structure: `{ pageAsset, content, nav, query }`.
- [ ] Comprehensive unit tests are implemented for all use cases and edge cases.
- [ ] Documentation in `README.md` provides clear guidance for usage and integration.

### Sub-Tasks & Estimates

1. Define the base GraphQL query constant.
2. Implement the `client.getPage` method.
3. Add validation for required inputs (`page`, `content`, or `nav`).
4. Parse `_map` fields in the response.
5. Write unit tests for the method.
6. Update the `README.md` with usage examples and details.

### Simple Use Cases
1. Basic Pages with Required Params

```js
const { pageAsset, query,  } = client.getPage({ url: '/home'  });
```

2. Basic Pages with Optional Params
```js
const { pageAsset, query } = client.getPage({url: '/home', language: "1", personaId: "personaId", site: ""});
```

### Advance Use Cases

1. Page with Content
```js
const { pageAsset, query, content } = await client.getPage({
    url: '/blog',
    content: {
        blogs: `search(query: "+contentType: blog", limit: 5) {
            title
            author {
                firstName
                lastName
            }
        }`
    }
});
```

2. Page With Navigation

```js
const { pageAsset, query, nav } = await client.getPage({
    url: '/services',
    nav: {
        menu: `search(query: "+contentType: menu", limit: 1) {
            items {
                title
                link
            }
        }`
    }
});
```

3. Page with Content and Navigation

```js
const data = await client.getPage({
    url: '/services',
    content: {
        services: `search(query: "+contentType: service", limit: 10) {
            title
            description
        }`
    },
    nav: {
        menu: `search(query: "+contentType: menu", limit: 1) {
            items {
                title
                link
            }
        }`
    }
});
```

4. Page with Page Fragments

```js
const data = await client.getPage({
    url: '/dashboard',
    pageFragment: `
        containers {
            containerContentlets {
                contentlets {
                    ... on Banner {
                        title
                        image {
                            url
                        }
                        link
                    }
                    ... on Widget {
                        widgetType
                        data {
                            key
                            value
                        }
                    }
                }
            }
        }
    `,
    content: {
        dashboardItems: `search(query: "+contentType: dashboardItem", limit: 5) {
            title
            description
            ...on DashboardItem {
                metrics {
                    name
                    value
                }
            }
        }`
    },
    nav: {
        sidebar: `search(query: "+contentType: sidebar", limit: 1) {
            items {
                title
                link
            }
        }`
    }
});
```

### Error Handling:

```js
async function fetchPageContent() {
    try {
        const data = await client.getPage({
            url: '/dashboard',
            content: {
                dashboardItems: `search(query: "+contentType: dashboardItem", limit: 5) {
                    badPropertyResultInErros
                }`
            },
        });

        return data;
    } catch (error) {
        console.error('Error fetching page content:', error);
        return null;
    }
}
```

#### Developer Note: Documentation for `getPageContent` Method

**Objective**: Ensure comprehensive documentation for the `getPageContent` method to clarify expected arguments and their purposes.

##### Key Points to Document:

1. **Arguments Expected**:
- **`url` (String)**: The only required property. It specifies the URL of the page to be fetched.
 - **`language` (String, Optional)**: Specifies the language in which the page content should be retrieved.
- **`pageFragment` (String, Optional)**: A query string to fetch specific fragments of the page, useful for modular content.

2. **Optional Properties**:
- **`content` (Object, Optional)**: 
  - This property is used to fetch additional content that does not belong to the page itself but is desired alongside the page. For example, related blog posts or promotional items.
  - Each key in the `content` object should correspond to a query that retrieves the desired data.

- **`nav` (Object, Optional)**: 
  - Used to fetch navigation-related data, such as menus, that complement the page layout.

3. **Error Handling**:
- Ensure that the documentation includes guidance on handling potential errors during the data retrieval process, as demonstrated in the provided error handling example.

##### Purpose of `content`:

The `content` argument is specifically designed to allow developers to fetch extra content that is not inherently part of the page's main structure. This enables the inclusion of dynamic or related content, enhancing the page's richness and user engagement.

### PR DESCRIPTION
**Pull request by rjvelazco**
# DotCMS SDK Migration Guide

This guide demonstrates the differences between the old and new SDK approaches. Each section shows the old way followed by the recommended new way.

## 1. Client Initialization

**❌ OLD WAY: Using static init method**
```typescript
const client = DotCmsClient.init({
    dotcmsUrl: 'https://demo.dotcms.com',
    authToken: '1234567890'
});
```

**✅ NEW WAY: Using factory function**
```typescript
const client = dotCMSCreateClient({
    dotcmsUrl: 'https://demo.dotcms.com',
    authToken: '1234567890'
});
```

## 2. Fetching Pages

**❌ OLD WAY: Using object parameter with path property**
```typescript
const page = client.page.get({ path: '/', depth: 1});
```

**✅ NEW WAY: Using direct path parameter**
```typescript
const page = client.page.get('/', { depth: 1 });
```

## 3. Fetching Navigation

**❌ OLD WAY: Using object parameter with path property**
```typescript
const navigation = client.nav.get({ path: '/', depth: 1, languageId: 1 });
```

**✅ NEW WAY: Using direct path parameter with options object**
```typescript
const { page, content, nav } = client.page.get("/", {
  languageId: "1",
  mode: "LIVE",
  query: {
    page: `
        containers {
          containerContentlets {
            uuid
            contentlets {
              title
            }
          }
        }
    `,
    content: {
      blogs: `
            query {
                blogs {
                    id
                    title
                    publishDate
                }
            }
        `,
    },
    nav: {
      mainNav: `
            query {
                nav {
                    items {
                        label
                        url
                    }
                }
            }
        `,
    },
  },
});
```

## 4. Working with Content

**❌ OLD WAY: Complex content queries**
```typescript
const content = client.content.query({
    query: 'contentType:Blog',
    limit: 10,
    offset: 0
});
```

**✅ NEW WAY: Simplified content queries**
```typescript
const content = client.content.query('contentType:Blog', {
    limit: 10,
    offset: 0
});
```

## 5. Editor Functionality

**❌ OLD WAY: Using editor property**
```typescript
// Access editor functionality
const editor = client.editor;
editor.on("edit", () => { /* YOUR CALLBACK */});
```

**✅ NEW WAY: Available in `@dotcms/uve`**

The editor functionality has been removed from the new SDK as it's being replaced by a separate package focused specifically on editing capabilities.

## Benefits of the New SDK

1. **More intuitive API**: Path parameters are passed directly instead of in objects
2. **Better TypeScript support**: Improved type definitions and error messages
3. **Enhanced functionality**: New features like personalized pages
4. **Consistent parameter patterns**: All methods follow the same pattern of (required, {options})
5. **Better error handling**: More descriptive error messages and proper error types
6. **Focused functionality**: Core API functionality separated from editor-specific features

This pull request includes significant updates to the `core-web/libs/sdk/client` module, focusing on refactoring and reorganizing the codebase. The changes involve deprecating old modules, introducing new client configurations, and adding comprehensive unit tests for the new functionalities.

### Refactoring and Deprecation:
* [`core-web/libs/sdk/client/project.json`](diffhunk://#diff-76e2a1ee1280d3b661ae48b02693271b4ae48e8924ddcf183a0dc4ddfb460715L29-R29): Updated the `entryFile` to point to the new location under `deprecated/editor`.
* [`core-web/libs/sdk/client/src/index.ts`](diffhunk://#diff-e1656e0e6ea2bbbf1cd43fb6b12f3e4c419d28dd7d11401f31facf5b3f4eb75aL1-R13): Refactored imports to use the new client structure and moved deprecated modules. [[1]](diffhunk://#diff-e1656e0e6ea2bbbf1cd43fb6b12f3e4c419d28dd7d11401f31facf5b3f4eb75aL1-R13) [[2]](diffhunk://#diff-e1656e0e6ea2bbbf1cd43fb6b12f3e4c419d28dd7d11401f31facf5b3f4eb75aL22-R23) [[3]](diffhunk://#diff-e1656e0e6ea2bbbf1cd43fb6b12f3e4c419d28dd7d11401f31facf5b3f4eb75aL52-R54)

### New Client Implementation:
* [`core-web/libs/sdk/client/src/lib/client/client.ts`](diffhunk://#diff-aa54238655bef05cdd48772e24046f2105dd74b1a20fac91e0b3ca610e3095a6R1-R153): Introduced `DotCMSClient` class and `dotCMSCreateClient` function to handle client initialization with proper configurations and sub-client initializations.

### Unit Tests:
* [`core-web/libs/sdk/client/src/lib/client/client.spec.ts`](diffhunk://#diff-8d6cdff4ac30eed476b3b4b42f31cd5ab77ecb0fcb37605bf9fd5e504b6c79a4R1-R158): Added unit tests for `DotCMSClient`, ensuring proper initialization, header handling, and validation.
* [`core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.spec.ts`](diffhunk://#diff-6efa73174333c167644002586b5144107991d3c681a0bd0e982aefe0eeeca7a5R1-R167): Added unit tests for `NavigationClient`, covering various scenarios including successful fetch, error handling, and parameter normalization.

### Navigation Client:
* [`core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.ts`](diffhunk://#diff-ccca05575b5095ec6e63c7a7d3be4723cc9ed76a5ddf12197c6ad456b5785dc7R1-R72): Implemented `NavigationClient` class to handle navigation-related API calls with support for depth and language parameters.

These changes collectively enhance the modularity, maintainability, and test coverage of the SDK client module.

This PR fixes: #30968

### CODE REVIEW

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+
+    /**
+     * Implementation of the get method that handles both REST API and GraphQL requests.
+     *
+     * @param {string} url - The URL of the page to retrieve
+     * @param {PageRequestParams | GraphQLPageOptions} [options] - Options for the request
+     * @returns {Promise<DotCMSPageAsset | {page: any, content: Record<string, any>, nav: Record<string, any>, errors?: any}>}
+     *          A Promise that resolves to the page data
+     */
+    async get(
```
Comment (fmontes) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:175]: I’m not convinced we should support this approach.

- It will create confusion for users.
- It may overcomplicate coding for us.
- If we can’t use GQL for all the cases, due to the ⁠_map issue, we should have a single method for retrieving a page.

We need:
- A unified API method to fetch a page.
- An option to decide whether to use GQL or not.

Additionally, I believe the new client shouldn’t have separate `⁠.content` and `⁠.nav`. Everything should be consolidated into one method only.

The way that I'm seeing this is that user will be bulding pages and the pages might have content to 1, 2 or 3 sources, but they shoudn't care about what source it is, they should just call one method and ask what they want and we deal with that internally.

And sorry I know we talk about this recently, but is deviating from the original idea of simplification and make it as complex as the `alpha` version if not more.

Comment (fmontes) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:175]: We might need to go back to the drawing board.

Comment (fmontes) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:175]: With this approach we have 2 of everying, 2 of get a page, 2 of get nav and 2 of get content, this is not simplifying.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    /**
+     * The depth of the page you want to retrieve.
+     */
+    depth?: string;
+
+    /**
+     * The publish date of the page you want to retrieve.
+     */
+    publishDate?: string;
+}
```
Comment (fmontes) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:87]:
```suggestion
/**
 * Maps frontend parameter types to backend string types
 */
type StringifyParam<T> = T extends string | number | boolean ? string : never;

/**
 * The private parameters for the Page API.
 * @internal
 */
export type BackendPageParams = {
    [K in keyof PageRequestParams as K extends 'siteId'
        ? 'hostId'
        : K extends 'languageId'
        ? 'language_id'
        : K extends 'personaId'
        ? 'com.dotmarketing.persona.id'
        : K]?: StringifyParam<PageRequestParams[K]>;
};
```

Of course I ai this type, I'm not †hat smart, but what I'm looking with this is keep the relationship of the two types.

What this is doing is creating a new type `BackendPageParams` by iterating trough the `PageRequestParams` and changing some keys based on the mapping.

`siteId --> hostId` and so on...

And the `StringifyParam<PageRequestParams[K]>`: Converts the value types to `string` to send to the backend.

Is just a suggestions, but we don't have to use it... I just think the relationship here is important.

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    }
+
+    /**
+     * Retrieves a page from DotCMS using REST API.
+     *
+     * @param {string} url - The URL of the page to retrieve
+     * @param {PageRequestParams} [options] - Options for retrieving the page via REST API
+     * @returns {Promise<DotCMSPageAsset>} A Promise that resolves to the page data
+     */
+    get(url: string, options?: PageRequestParams): Promise<DotCMSPageAsset>;
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:155]: We need to add examples on the jsdoc, as user idk if the path is with "/" or not

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+        options?: PageRequestParams | GraphQLPageOptions
+    ): Promise<DotCMSPageAsset | DotCMSGraphQLPageResponse> {
+        const isGraphQLRequest = !!(options as GraphQLPageOptions)?.['query'];
+
+        if (isGraphQLRequest) {
+            return this.getPageFromGraphQL(url, options as GraphQLPageOptions);
+        }
+
+        return this.getPageFromAPI(url, options);
+    }
```
[the remark this answers is not in the document]
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:186]: Added!

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts (hunk 2 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts]: Good catch, i added a mapping for this

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+
+        const requestHeaders = this.requestOptions.headers as Record<string, string>;
+        const requestBody = JSON.stringify({ query: completeQuery, variables: requestVariables });
+
+        try {
+            const { data, errors } = await fetchGraphQL({
+                body: requestBody,
+                headers: requestHeaders
+            });
+
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:387]: we want to throw an Error here? The api response is a 200 with { errors }, but maybe we want to have a similar error handling as the REST endpoint

File: core-web/libs/sdk/client/src/lib/client/client.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+ *
+ * @param url - The URL string to parse
+ * @returns A URL object if parsing is successful, undefined otherwise
+ */
+function parseUrl(url: string): URL | undefined {
+    try {
+        return new URL(url);
+    } catch {
+        return undefined;
+    }
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/client.ts:50]: We should do a `console.error` here

File: core-web/libs/sdk/client/src/lib/client/client.ts [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+ *
+ * // Use the client to fetch content
+ * const pages = await client.page.get('/about-us');
+ * ```
+ */
+export const createDotCMSClient = (clientConfig: DotCMSClientConfig): DotCMSClient => {
+    const { dotcmsUrl, authToken } = clientConfig || {};
+    const instanceUrl = parseUrl(dotcmsUrl)?.origin;
+
+    if (!instanceUrl) {
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/client.ts:138]: I think by adding the console in the `parseUrl` function and also logging the failed value, is more meaningful and cleaner

File: core-web/libs/sdk/client/src/lib/client/client.ts (hunk 6 above)
Comment (zJaaal) [reply core-web/libs/sdk/client/src/lib/client/client.ts:50]: Also rename to `parseURL` to maintain consistency

File: core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    async get(path: string, params?: NavRequestParams): Promise<unknown> {
+        if (!path) {
+            throw new Error("The 'path' parameter is required for the Navigation API");
+        }
+
+        const navParams = params ? this.mapToBackendParams(params) : {};
+        const urlParams = new URLSearchParams(navParams).toString();
+
+        const pathWithoutSlash = path.replace(/^\//, '');
+        const url = `${this.BASE_URL}/${pathWithoutSlash}${urlParams ? `?${urlParams}` : ''}`;
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.ts:46]: We should not be doing this, this breaks vanity URL, take the path as it is and build the URL with the URL API

Comment (zJaaal) [reply core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.ts:46]: At most, you can clean double `/`

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 9]
```diff
@@ -0,0 +1,441 @@
...
+export interface PageRequestParams {
+    /**
+     * The id of the site you want to interact with. Defaults to the one from the config if not provided.
+     */
+    siteId?: string;
+
+    /**
+     * The mode of the page you want to retrieve. Defaults to the site's default mode if not provided.
+     */
+    mode?: 'EDIT_MODE' | 'PREVIEW_MODE' | 'LIVE';
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:20]: It's time to centralize this enum and use it

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+     */
+    publishDate?: string;
+}
+
+export interface GraphQLPageOptions extends PageRequestParams {
+    query: {
+        page?: string;
+        content?: Record<string, string>;
+        nav?: Record<string, string>;
+        variables?: Record<string, string>;
```
[the remark this answers is not in the document]
Comment (zJaaal) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:71]: That should be sent, inside the page, so the page maybe should be an object.

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 11]
```diff
@@ -0,0 +1,441 @@
...
+            'siteId',
+            'mode',
+            'languageId',
+            'personaId',
+            'fireRules',
+            'depth',
+            'publishDate'
+        ];
+
+        return Object.keys(options).every((key) => validKeys.includes(key));
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:258]: This will fail if I wrongly send more `queryParams` so it probably will fail inside UVE, because we send experiments and date params

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts [hunk 12]
```diff
@@ -0,0 +1,441 @@
...
+     */
+    async #getPageFromAPI(path: string, params?: PageRequestParams): Promise<DotCMSPageAsset> {
+        if (!path) {
+            throw new Error("The 'path' parameter is required for the Page API");
+        }
+
+        const pagePath = path.replace(/^\//, '');
+        const pageParams = this.#mapToBackendParams(params || {});
+        const urlParams = new URLSearchParams(pageParams as Record<string, string>).toString();
+        const url = `${this.BASE_URL}/json/${pagePath}${urlParams ? `?${urlParams}` : ''}`;
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/page-api.ts:294]: Same here as my comment about the path above

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts (hunk 5 above)
Comment (zJaaal) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:387]: For sure, if it has errors we should throw one, so they can catch it

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+ * @throws {Error} If the HTTP response is not successful
+ */
+export async function fetchGraphQL({
+    body,
+    headers
+}: {
+    body: string;
+    headers: Record<string, string>;
+}) {
+    const response = await fetch('http://localhost:8080/api/v1/graphql', {
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/utils.ts:206]: Beware of this

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+        method: 'POST',
+        body,
+        headers
+    });
+
+    if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/utils.ts:213]: We can do this error, better, it does not give any context of where it happened

File: core-web/libs/sdk/client/src/lib/deprecated/editor/models/client.model.ts [hunk 15]
```diff
@@ -107,3 +107,18 @@ export interface DotUVE {
...
+
+/**
+ * Represents a listener for DotcmsClientListener.
+ *
+ * @typedef {Object} DotcmsClientListener
+ * @property {string} action - The action that triggers the event.
+ * @property {string} event - The name of the event.
+ * @property {function(...args: any[]): void} callback - The callback function to handle the event.
+ */
+export type DotcmsClientListener = {
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/deprecated/editor/models/client.model.ts:119]: Where is this being used?

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts (hunk 11 above)
Comment (zJaaal) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:258]: A `some` instead of `every` should do the trick

File: core-web/libs/sdk/client/src/lib/client/client.ts (hunk 7 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/client.ts:138]: I add the console.error on parseURL, but we need to check here the instanceURL for type checking in new DotCMSClient(config)

File: core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.ts (hunk 8 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/navigation/navigation-api.ts]: Using the URL API didn't help me achieve what we were looking for, what I did was remove the repeated slashes

File: core-web/libs/sdk/client/src/lib/deprecated/editor/models/client.model.ts (hunk 15 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/deprecated/editor/models/client.model.ts:119]: The old DotcmsClient have a listeners object with this type, to save and unsubscribe all event listeners

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts (hunk 9 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:21]: This should probably be inside `@dotcms/uve/types`, 
We have a ticket that separates the logic of both libraries, this change can go there

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts [hunk 16]
```diff
@@ -138,17 +159,20 @@ export const buildPageQuery = (pageQuery: string, additionalQueries: string) =>
     }
   }
 
-  ${pageQuery ? ` fragment ClientPage on DotPage { ${pageQuery} } ` : ''}
+  ${page ? ` fragment ClientPage on DotPage { ${page} } ` : ''}
+
+  ${fragments ? fragments.join(' ') : ''}
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/utils.ts:164]: use `\n` or `\n\n` to make it readable

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts [hunk 17]
```diff
@@ -197,20 +221,27 @@ export function mapResponseData(
...
+    baseURL,
     body,
     headers
 }: {
+    baseURL: string;
     body: string;
     headers: Record<string, string>;
 }) {
-    const response = await fetch('http://localhost:8080/api/v1/graphql', {
+    const response = await fetch(`${baseURL}/api/v1/graphql`, {
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/utils.ts:232]: We should create URLs with URL API, to prevent unhandled errors

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/client/src/lib/client/page/page-api.ts (hunk 11 above)
Comment (zJaaal) [reply core-web/libs/sdk/client/src/lib/client/page/page-api.ts:266]: By checking with `some` method, we should be safe

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts (hunk 17 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/page/utils.ts]: Added!

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts (hunk 16 above)
Comment (KevinDavilaDotCMS) [reply core-web/libs/sdk/client/src/lib/client/page/utils.ts]: Added!

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/sdk/client/src/lib/client/page/utils.ts [hunk 18]
```diff
@@ -0,0 +1,251 @@
...
+    baseURL,
+    body,
+    headers
+}: {
+    baseURL: string;
+    body: string;
+    headers: Record<string, string>;
+}) {
+    const url = new URL(baseURL);
+    url.pathname = '/api/v1/graphql';
```
Comment (zJaaal) [inline core-web/libs/sdk/client/src/lib/client/page/utils.ts:233]: Awesome!

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]
